misaligned.cpp

```diff
@@ -1,21 +1,69 @@
 #include <assert.h>
 #include <iostream>
+#include <string>
+#include <vector>
 
-int printColorMap() {
-    const char* majorColor[] = {"White", "Red", "Black", "Yellow", "Violet"};
-    const char* minorColor[] = {"Blue", "Orange", "Green", "Brown", "Slate"};
+
+const char* majorColor[] = { "White", "Red", "Black", "Yellow", "Violet" };
+int noOfMajorColor = sizeof(majorColor) / sizeof(majorColor[0]);
+const char* minorColor[] = { "Blue", "Orange", "Green", "Brown", "Slate" };
+int noOfMinorColor = sizeof(majorColor) / sizeof(minorColor[0]);
+
+
+std::vector<std::string> tokenize(const std::string& str, const std::string& token) {
+    size_t start = str.find_first_not_of(token), end = start;
+    std::vector<std::string> token_v;
+    while (start != std::string::npos) {
+        // Find next occurence of delimiter
+        end = str.find(token, start);

```

Comment:
Doesn't this assume there will always be a delimiter at the end of the `str`?
Whenever I write code for a problem that may already have been solved, I ask ChatGPT / copilot once.